test/gtest/conv_immediate.cpp

```diff
@@ -0,0 +1,532 @@
+        }
+
+        if(max_diff > integer_interval)
+            valid_result = false;
+    }
+
+    if(!valid_result)
+    {
+        std::cout << "diff at:" << idx << ", gpu:" << t_gpu[idx] << ", cpu:" << t_cpu[idx]
+                  << std::endl;
+    }
+    return valid_result;
+}
+} // namespace
+
+template <class Types>
+class ReferenceConvBase : public ::testing::Test
+{
+protected:
+    virtual void SetUp() override {}
+
+    virtual void TearDown() override {}
+
+    static std::vector<int> get_image_depth() { return {8, 10}; }
+
+    static std::vector<int> get_image_size() { return {9, 14}; }
+
+    static std::vector<int> get_filter_depth() { return {1, 3}; }
+
+    static std::vector<int> get_filter_size() { return {1, 3}; }
+
+    static std::vector<int> get_stride_depth() { return {1, 2}; }
+
+    static std::vector<int> get_dilation_depth() { return {1}; }
+
+    static std::vector<int> get_stride_dilation_size() { return {1, 2}; }
+
+    static std::vector<int> get_pad_depth() { return {0, 1}; }
+
+    static std::vector<int> get_pad_size() { return {0, 1}; }
+
+    static std::vector<int> get_group_size() { return {1, 2}; }
+
+    static std::vector<int> get_channel_size()
+    {
+        // Channel size must be multiple of group size
+        auto sz = get_group_size();
+        std::transform(sz.begin(), sz.end(), sz.begin(), [&](int s) { return s * 4; });
+
+        return sz;
+    }
+
+    static std::vector<int> get_batch_size() { return {1, 2}; }
+
+    static bool same(std::vector<int> v) { return std::equal(v.begin() + 1, v.end(), v.begin()); }
+
+    template <typename F>
+    void iterate_conv_2d(F f)
+    {
+        using namespace test;
+        auto test_cases = cartesian_product_axx(
+            get_channel_size(), get_image_size(), get_filter_size(), get_pad_size());
+
+        for(auto test_case : test_cases)
+        {
+            int c, hi, wi, fy, fx, py, px;
+            std::tie(c, hi, wi, fy, fx, py, px) = test_case;
+
+            int n = get_batch_size()[prng::gen_canonical<size_t>()];
+            int g = get_group_size()[prng::gen_canonical<size_t>()];
+            int k = get_channel_size()[prng::gen_canonical<size_t>()];
+
+            int sy = get_stride_dilation_size()[prng::gen_canonical<size_t>()];
+            int sx = get_stride_dilation_size()[prng::gen_canonical<size_t>()];
+
+            int dy = get_stride_dilation_size()[prng::gen_canonical<size_t>()];
+            int dx = get_stride_dilation_size()[prng::gen_canonical<size_t>()];
+
+            int ho = conv_out_size(hi, py, dy, fy, sy);
+            int wo = conv_out_size(wi, px, dx, fx, sx);
+
+            if(fy > hi || fx > wi || (fy - 1) < py || (fx - 1) < px || ho <= 0 || wo <= 0 ||
+               c % g != 0 || k % g != 0)
+                continue;
+            if((fx == 3 && fy == 5) || (fx == 5 && fy == 3))
+                continue;
+
+            f(n, {wi, hi}, c, k, {fy, fx}, {py, px}, {sy, sx}, {dy, dx}, g);
+        }
```

Comment:
Valid point, but the overhead associated with those test cases is horrible for a test space this vast (it was over 80k test cases at one point). I usually add custom instrumentation anyway, so adding a filter or custom data to isolate the bad ones before rebuilding is trivial.

Reply:
Sure, not a big deal. May I ask you to investigate that horrible overhead and provide some recommendations when each approach should be used and why?

Reply:
That investigation is far from trivial, so (just to be clear) it will not be part of this PR. There was another team member who pointed this out not long after I saw it so there may already be a ticket (@BrianHarrisonAMD maybe?). If not, I'll make a ticket and we can discuss priority during planning.

Reply:
I don't think I ever added anything.

Are you saying that the testing combine macros is leading to the delay when executing tests through make check? 
I think we see this delay even if they should be skipped.

Oddly, I don't notice any delay when executing the tests through the executables that are generated.
I would have expected them to behave the same.

Reply:
Then we postpone this PR until that investigation has not been done. We can't just randomly do various things. We've already got the test broken due to this.